Documentation/Specification/specification.md

```diff
@@ -71,34 +106,95 @@ Below 10 digits, only even numbers are valid lengths.
 
 The default length for most purposes is 10 digits.
 
-### Formatting
+### Separator And Padding
 
-The format separator must be inserted after eight digits.
-If the requested code length is fewer than eight digits, the remaining digits before the format separator must consist of the padding character.
+The format separator "+" must be inserted after eight digits.
+If the requested code length is fewer than eight digits, the remaining digits before the format separator must consist of the padding character "0".
 
-### Code precision
+## Encoding Algorithms
 
-The following table gives the precision of the valid code lengths in degrees and in meters. Where the precisions differ between latitude and longitude both are shown:
+It is recommended for any encoding algorithm to convert the input latitude and longitude coordinates into integer data types.
+Floating-point types should be avoided due to the limitations of accurately representing floating-point numbers.
+(IEEE 754 implementations result in incorrect codes being generated.)
+Implementations using high precision floating-point libraries should be well tested to ensure correct codes are returned.
 
-| Code length | Precision in degrees | Precision        |
-| :---------: | :------------------: | :--------------: |
-| 2           | 20                   | 2226 km          |
-| 4           | 1                    | 111.321 km       |
-| 6           | 1/20                 | 5566 meters      |
-| 8           | 1/400                | 278 meters       |
-| 10          | 1/8000               | 13.9 meters      |
-| 11          | 1/40000 x 1/32000    | 2.8 x 3.5 meters |
-| 12          | 1/200000 x 1/128000  | 56 x 87 cm       |
-| 13          | 1/1e6 x 1/512000     | 11 x 22 cm       |
-| 14          | 1/5e6 x 1/2.048e6    | 2 x 5 cm         |
-| 15          | 1/2.5e7 x 1/8.192e6  | 4 x 14 mm        |
+The algorithms below should be considered as reference implementations.
+Other implementations should be well tested to make sure they do not differ.
 
-NB: This table assumes one degree is 111321 meters, and that all distances are calculated at the equator.
+Whether you build the code from the first digit or the last digit may depend on personal preference, or the available string-building libraries.
+
+### Prerequisites
+
+Both the encoding algorithms below operate on integer values, requiring the latitude and longitude values to be converted to be positive integers.
+Take care when implementing this conversion - merging the operations below can result in unexpected rounding operations and will result in test failures.
+
+1. Convert latitude to a clipped, positive integer:
+   1. Multiply it by the maximum latitude resolution, 2.5e7, and truncate (_not round_) it.
+   2. Add `90 x 2.5e7` to put it in a positive range.
+   3. Clip it so that it is in the range `0 <= latitude < 180`. (If it is greater than or equal to 180, set it to `180 - 1/8.192e6`.)
```

Comment:
Since we multiply by a large constant before, the maximum here should probably be `180 x 2.5e7`?